### PR TITLE
VSCode: Java TextMate grammar updated.

### PR DESCRIPTION
**Pull request by dbalek**
Java TextMate grammar updated to recent version.

### CODE REVIEW

**Review by MartinBalin — APPROVED** [review]

File: java/java.lsp.server/vscode/syntaxes/java.tmLanguage.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
 					"name": "invalid.deprecated.package_name_not_lowercase.java"
 				},
 				{
```
Comment (sdedic) [inline java/java.lsp.server/vscode/syntaxes/java.tmLanguage.json:42]: `syncronized` ... a typo ?

Comment (dbalek) [reply java/java.lsp.server/vscode/syntaxes/java.tmLanguage.json:42]: File is copied from `https://github.com/microsoft/vscode/blob/main/extensions/java/syntaxes/java.tmLanguage.json`

File: java/java.lsp.server/vscode/syntaxes/java.tmLanguage.json [hunk 2]
```diff
@@ -1571,42 +1582,45 @@
...
 					"end": "\"",
 					"endCaptures": {
 						"0": {
-							"name": "punctuation.definition.string.end.java string.quoted.double.java"
+							"name": "punctuation.definition.string.end.java"
 						}
 					},
+					"name": "string.quoted.double.java",
 					"patterns": [
-						{
```
Comment (lahodaj) [inline java/java.lsp.server/vscode/syntaxes/java.tmLanguage.json:1588]: Sorry, not sure about this removal. Please see:
https://github.com/apache/netbeans/pull/6745

**Review by lahodaj — COMMENTED** [review]
Please see an inline comment.

File: java/java.lsp.server/vscode/syntaxes/java.tmLanguage.json (hunk 2 above)
Comment (dbalek) [reply java/java.lsp.server/vscode/syntaxes/java.tmLanguage.json:1588]: Injection created for string templates.

**Review by sdedic — APPROVED** [review]
OK based on my poor TextMate knowledge.

**Review by lahodaj — APPROVED** [review]
I am not quite sure how it works, but I tried, and it seems to work. So, OK from me.

**Pull request merged**